[BUG] ci build failing with /usr/include/c++/12/type_traits update in jammy

### Describe the bug
<!--- A clear description of what the bug is. -->

Working
```
docker@22fe29c36e49:/var/media/DATA/home-rudi/LibreELEC.kernel11$ ls -la /usr/include/c++/12/type_traits
-rw-r--r-- 1 root root 110398 May 13  2022 /usr/include/c++/12/type_traits
```
not working

```
docker@22fe29c36e49:/var/media/DATA/home-rudi/LibreELEC.kernel11$ ls -la /usr/include/c++/12/type_traits
-rw-r--r-- 1 root root 110506 May 13 07:33 /usr/include/c++/12/type_traits
```

```
[ 89%] Building CXX object src/CMakeFiles/ccache_framework.dir/util/file.cpp.o
In file included from /var/media/DATA/home-rudi/LibreELEC.kernel11/build.LibreELEC-Generic.x86_64-12.0-devel/build/ccache-4.8.2/src/storage/local/LocalStorage.cpp:25:
/var/media/DATA/home-rudi/LibreELEC.kernel11/build.LibreELEC-Generic.x86_64-12.0-devel/build/ccache-4.8.2/src/storage/local/LocalStorage.cpp: In instantiation of 'storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)> [with auto:39 = unsigned char; auto:40 = std::function<void(double)>]':
/usr/include/c++/12/type_traits:2559:26:   required by substitution of 'template<class _Fn, class ... _Args> static std::__result_of_success<decltype (declval<_Fn>()((declval<_Args>)()...)), std::__invoke_other> std::__result_of_other_impl::_S_test(int) [with _Fn = storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>&; _Args = {unsigned char, const std::function<void(double)>&}]'
/usr/include/c++/12/type_traits:2570:55:   required from 'struct std::__result_of_impl<false, false, storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>&, unsigned char, const std::function<void(double)>&>'
/usr/include/c++/12/bits/std_function.h:348:9:   recursively required by substitution of 'template<class _Result, class _Ret> struct std::__is_invocable_impl<_Result, _Ret, true, std::__void_t<typename _CTp::type> > [with _Result = std::__invoke_result<storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>&, unsigned char, const std::function<void(double)>&>; _Ret = void]'
/usr/include/c++/12/bits/std_function.h:348:9:   required from 'struct std::function<void(unsigned char, const std::function<void(double)>&)>::_Callable<storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>, storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>, std::__invoke_result<storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>&, unsigned char, const std::function<void(double)>&> >'
/usr/include/c++/12/bits/std_function.h:353:8:   required by substitution of 'template<class _Res, class ... _ArgTypes> template<class _Cond, class _Tp> using _Requires = std::__enable_if_t<_Cond::value, _Tp> [with _Cond = std::function<void(unsigned char, const std::function<void(double)>&)>::_Callable<storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>, storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>, std::__invoke_result<storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>&, unsigned char, const std::function<void(double)>&> >; _Tp = void; _Res = void; _ArgTypes = {unsigned char, const std::function<void(double)>&}]'
/usr/include/c++/12/bits/std_function.h:434:9:   required by substitution of 'template<class _Functor, class _Constraints> std::function<void(unsigned char, const std::function<void(double)>&)>::function(_Functor&&) [with _Functor = storage::local::LocalStorage::recompress(std::optional<signed char>, uint32_t, const storage::local::ProgressReceiver&)::<lambda(const auto:39&, const auto:40&)>; _Constraints = <missing>]'
/var/media/DATA/home-rudi/LibreELEC.kernel11/build.LibreELEC-Generic.x86_64-12.0-devel/build/ccache-4.8.2/src/storage/local/LocalStorage.cpp:701:24:   required from here
/var/media/DATA/home-rudi/LibreELEC.kernel11/build.LibreELEC-Generic.x86_64-12.0-devel/build/ccache-4.8.2/src/Logging.hpp:33:7: internal compiler error: Segmentation fault
   33 |       Logging::log(std::string_view(message_));                                \
      |       ^~~~~~~
/var/media/DATA/home-rudi/LibreELEC.kernel11/build.LibreELEC-Generic.x86_64-12.0-devel/build/ccache-4.8.2/src/Logging.hpp:39:27: note: in expansion of macro 'LOG_RAW'
   39 | #define LOG(format_, ...) LOG_RAW(fmt::format(FMT_STRING(format_), __VA_ARGS__))
      |                           ^~~~~~~
/var/media/DATA/home-rudi/LibreELEC.kernel11/build.LibreELEC-Generic.x86_64-12.0-devel/build/ccache-4.8.2/src/storage/local/LocalStorage.cpp:710:13: note: in expansion of macro 'LOG'
  710 |             LOG("Failed to acquire content lock for {}/{}", l1_index, l2_index);
      |             ^~~
0x1b3ed3a internal_error(char const*, ...)
        ???:0
0x1013433 walk_tree_1(tree_node**, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*, tree_node* (*)(tree_node**, int*, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*))
        ???:0
0x101362e walk_tree_1(tree_node**, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*, tree_node* (*)(tree_node**, int*, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*))
        ???:0
0x1013433 walk_tree_1(tree_node**, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*, tree_node* (*)(tree_node**, int*, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*))
        ???:0
0x10135e4 walk_tree_1(tree_node**, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*, tree_node* (*)(tree_node**, int*, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*))
        ???:0
0x10135e4 walk_tree_1(tree_node**, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*, tree_node* (*)(tree_node**, int*, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*))
        ???:0
0x1013668 walk_tree_1(tree_node**, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*, tree_node* (*)(tree_node**, int*, tree_node* (*)(tree_node**, int*, void*), void*, hash_set<tree_node*, false, default_hash_traits<tree_node*> >*))
        ???:0
0x802c5e check_for_bare_parameter_packs(tree_node*, unsigned int)
        ???:0
0x843619 finish_expr_stmt(tree_node*)
        ???:0
0x82e85d tsubst_lambda_expr(tree_node*, tree_node*, int, tree_node*)
        ???:0
0x813675 tsubst_copy_and_build(tree_node*, tree_node*, int, tree_node*, bool, bool)
        ???:0
0x811d90 tsubst_copy_and_build(tree_node*, tree_node*, int, tree_node*, bool, bool)
        ???:0
0x8225f9 instantiate_decl(tree_node*, bool, bool)
        ???:0
0x738e17 mark_used(tree_node*, int)
        ???:0
0x6bfa5b build_op_call(tree_node*, vec<tree_node*, va_gc, vl_embed>**, int)
        ???:0
0x848447 finish_call_expr(tree_node*, vec<tree_node*, va_gc, vl_embed>**, bool, bool, int)
        ???:0
0x811fe6 tsubst_copy_and_build(tree_node*, tree_node*, int, tree_node*, bool, bool)
        ???:0
0x81b12f tsubst(tree_node*, tree_node*, int, tree_node*)
        ???:0
0x827f0d tsubst_template_args(tree_node*, tree_node*, int, tree_node*)
        ???:0
0x81acaa tsubst(tree_node*, tree_node*, int, tree_node*)
        ???:0
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <file:///usr/share/doc/gcc-12/README.Bugs> for instructions.
make[2]: *** [src/CMakeFiles/ccache_framework.dir/build.make:580: src/CMakeFiles/ccache_framework.dir/storage/local/LocalStorage.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:306: src/CMakeFiles/ccache_framework.dir/all] Error 2
make: *** [Makefile:156: all] Error 2n
```

### To Reproduce
Steps to reproduce the behavior:
1. Go to '...'
2. Play '....'
3. See error

### Informations
 - LE Version: [e.g. 9.2.1]
 - Hardware Platform: [e.g. RPi3]

### Log file
<!-- Add log files (https://libreelec.wiki/how_to/provide_logfile) that we can search for errors. -->

### Additional context
<!--- Add any other context about the problem here. -->

--- a/packages/devel/ccache/patches/ccache-0001-work-around-gcc-12-3-bug-109241.patch
+++ b/packages/devel/ccache/patches/ccache-0001-work-around-gcc-12-3-bug-109241.patch
@@ -1,0 +1,25 @@
+From 689168c292f1ed26c5f4a3070aeb649dad7facb5 Mon Sep 17 00:00:00 2001
+From: Joel Rosdahl <joel@rosdahl.net>
+Date: Tue, 1 Aug 2023 12:30:12 +0200
+Subject: [PATCH] fix: Work around GCC 12.3 bug 109241
+
+See also #1289.
+---
+ src/storage/local/LocalStorage.cpp | 4 +++-
+ 1 file changed, 3 insertions(+), 1 deletion(-)
+
+diff --git a/src/storage/local/LocalStorage.cpp b/src/storage/local/LocalStorage.cpp
+index 3e5abe5074..d0a904e553 100644
+--- a/src/storage/local/LocalStorage.cpp
++++ b/src/storage/local/LocalStorage.cpp
+@@ -854,7 +854,9 @@ LocalStorage::recompress(const std::optional<int8_t> level,
+           auto l2_content_lock = get_level_2_content_lock(l1_index, l2_index);
+           l2_content_lock.make_long_lived(lock_manager);
+           if (!l2_content_lock.acquire()) {
+-            LOG("Failed to acquire content lock for {}/{}", l1_index, l2_index);
++            // LOG_RAW+fmt::format instead of LOG due to GCC 12.3 bug #109241
++            LOG_RAW(fmt::format(
++              "Failed to acquire content lock for {}/{}", l1_index, l2_index));
+             return;
+           }
+ 

